Udf dimensionality reduction (PCA & T-SNE)

JeroenWannijn opened the pull request:
Includes https://github.com/Open-EO/openeo-community-examples/tree/test-apply-metadata since apply metadata is also included

HansVRP commented on python/DimensionalityReduction/udf_dim_reduction.py:
```diff
@@ -0,0 +1,209 @@
+    file_name = os.path.basename(urlparse(url).path)
+    file_path = os.path.join(cache_dir, file_name)
+
+    if os.path.exists(file_path):
```
we should check with the core team if this makes sence. Is this now donwloaded on an executor? ideally we download to the driver. 

This can be done automatically using the UDF dpeendency archive job options; which do require you to zip the actual model

JeroenWannijn replied:
I now give the model (zipped) as a job option to the job and it will consequently load the model from the working-directory from the driver. I added a script that search the model on the working-directory, in case something goes wrong when deploying dependencies

HansVRP commented on python/DimensionalityReduction/udf_dim_reduction.py:
```diff
@@ -0,0 +1,209 @@
+    transformed_normalized = scaler.fit_transform(transformed)
+    
+    # Reshape back to (n_components, y, x)
+    result_data = transformed_normalized.T.reshape((n_components, y, x))
```
would be good to check with Lori; now we use the new components; I believe the typical workflow is to use PCA and Tsne to find out the orginal best components and only use those

JeroenWannijn replied:
I added it after discussion with Lori, the results do look better

HansVRP reviewed (changes requested):
FOr the community notebook I would build of dependence from EOprocessing; 

Can we create a simple alternative where we create a big cube ourselves with standard openEO code?

I would remove the run locally and just keep the cloud based one.

Ideally we have 2? UDFS
1) which calculates the PCA and goes from cube[B01_min, B01_mean, B01_max .... B12_min, B12_mean, B12_max ....] --> cube[PCA1, PCA2...]

2)  cube[PCA1, PCA2...] --> cube[B01_min, B12_mean, B12_max] which then only keeps to most relevant actual bands

JeroenWannijn commented:
Changes for new review: 
- removed local run
- created my own master data cube
- added cloud mask -> improved the results the most
- added dependencies from job options and removed them from the job context
- added feature selection udf to select significant bands
